lib/maildir/unique_name.rb

```diff
@@ -7,7 +7,7 @@ class << self
     def counter
       @counter_mutex ||= Mutex.new
       @counter_mutex.synchronize do
-        @counter = @counter.to_i + 1
+        @counter = (@counter ||=  0) + 1
```

Comment:
`nil.to_i == 0` on every Ruby interpreter I know. Other than a coding style issue, what's the benefit here? It's not the method's job to initialize @counter; it just needs to make sure that incrementing the counter doesn't raise an exception.